### PR TITLE
Domain based config

### PR DESCRIPTION
**Pull request by somebody32**
Hi, 

I've added opportunity to specify locale per domain or subdomain. Just specify config hash like this: 

``` ruby
LocaleSetter::Domain.localized_domains = {
  "en.domain.com" => :en,
  "es.domain.com" => :es
}
```

and all should work as expected. 

Also cleaned up dummy app a little bit and updated gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,4 +15,5 @@ spec/reports
 test/tmp
 test/version_tmp
 tmp
-spec/dummy/log
+spec/dummy/log/*
```
Comment (steveklabnik) [inline .gitignore]: are these two lines not equivalent?

File: README.md [hunk 2]
```diff
@@ -139,6 +141,19 @@ LocaleSetter::User.locale_method = :my_locale
...
+You could specify prefered locale according to the domain (or subdomain). Just specify config hash like this:
+
+```ruby
+LocaleSetter::Domain.localized_domains = {
+  "en.domain.com" => :en,
+  "es.domain.com" => :es
+}
+```
+
+
```
Comment (steveklabnik) [inline README.md]: :scissors: these extra lines please. Just one between the backticks and the next header.

File: spec/locale_setter/domain_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    it "ignores a blank domain" do
+      described_class.for("", ["domain_locale"]).should_not be
+    end
+
+    it "ignores domains not from the list" do
+      described_class.for("my_domain.com", ["default"]).should_not be
+    end
+
+    it "detects right domain" do
+      described_class.for("my_domain.com", ["domain_locale"]).should == :domain_locale
```
Comment (steveklabnik) [inline spec/locale_setter/domain_spec.rb]: I know the rest of the codebase does not yet use it, but please change these tests into the `expect().to` style.

**Comment by steveklabnik** [comment]
Thanks for this! Overall, pretty good. Let's just address these few comments and I'm good to merge.

**Comment by somebody32** [comment]
Thanks for reviewing, I've made the changes

File: spec/locale_setter/generic_spec.rb [hunk 4]
```diff
@@ -42,6 +42,15 @@
       end
     end
 
+    context "with localized domains" do
+      it "uses the locale from config" do
+        i18n.available_locales = [:en, :domain_specified]
+        LocaleSetter::Domain.localized_domains = { "localized_domain.com" => :domain_specified }
+        setter.set_locale(i18n, {:domain => "localized_domain.com"} )
+        i18n.locale.should == :domain_specified
```
Comment (steveklabnik) [inline spec/locale_setter/generic_spec.rb]: Sorry, there's one more `should` here.

**Comment by somebody32** [comment]
I've got rid of shoulds for the whole spec file

**Comment by steveklabnik** [comment]
Thanks! :D

**Pull request merged**